### PR TITLE
[LLM Runtime] enable MHA fusion for gptneox&dolly&starcoder&llama2-70b

### PR DESCRIPTION
**Pull request by intellinjun**
## Type of Change

feature or bug fix or documentation or others
API changed or not:not

enable MHA fusion for gptneox&dolly&starcoder&llama2-70b
## Description

detail description 
JIRA ticket: xxx

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Comment by intellinjun** [comment]
https://inteltf-jenk.sh.intel.com/job/ITREX-cpp-graph-extension/158/

**Comment by intellinjun** [comment]
image

**Comment by intellinjun** [comment]
image

**Comment by intellinjun** [comment]
image

**Comment by intellinjun** [comment]
image

**Review by DDEle — APPROVED** [review]
LGTM

**Comment by airMeng** [comment]
can you summary the comparison, before and after?

**Review by airMeng — APPROVED** [review]

**Review by a32543254 — APPROVED** [review]
LGTM

File: intel_extension_for_transformers/llm/runtime/graph/models/llama/llama.cpp [hunk 1]
```diff
@@ -264,17 +264,18 @@ static bool llama_model_eval_internal(model_context& lctx, const model_token* to
...
-                                        0, 0,                      // nb (jblas managed)
-                                        il * v_size);              // offset
+        const auto v_cache = ne_view_3d(ctx0, kv_self.v,              // tensor
+                                        head_size, n_ctx, n_head_kv,  // ne
+                                        0, 0,                         // nb (jblas managed)
+                                        il * v_size);                 // offset
         // jblas alway view V as (D, n_head, seq)
-        const auto Vcur_plain = ne_reshape_3d(ctx0, ne_view_1d(ctx0, Vcur, n_embd * N, 0), n_embd / n_head, n_head, N);
+        const auto Vcur_plain =
+            ne_reshape_3d(ctx0, ne_view_1d(ctx0, Vcur, n_embd_gqa * N, 0), n_embd_gqa / n_head_kv, n_head_kv, N);
```
[the remark this answers is not in the document]
Comment (intellinjun) [reply intel_extension_for_transformers/llm/runtime/graph/models/llama/llama.cpp:278]: done

**Comment by intellinjun** [comment]
> can you summary the comparison, before and after?

I used ci test to get the summary,there will be results tomorrow.

**Comment by intellinjun** [comment]
starcoder still has accuracy issues, please don't merge

**Comment by intellinjun** [comment]
https://[inteltf-jenk.sh.intel.com/job/ITREX-cpp-graph-extension/169/artifact/deploy/engine/mistral-7b/mistral-7b-q4_j_b128-48-1-32-32.log](https://inteltf-jenk.sh.intel.com/job/ITREX-cpp-graph-extension/169/artifact/deploy/engine/mistral-7b/mistral-7b-q4_j_b128-48-1-32-32.log)

**Pull request merged**